calyx/src/passes/well_formed.rs

```diff
@@ -78,6 +114,70 @@ where
     Ok(())
 }
 
+fn same_binding(
+    binding_out: &SmallVec<[(ir::Id, u64); 5]>,
+    binding_in: &SmallVec<[(ir::Id, u64); 5]>,
+) -> CalyxResult<()> {
+    if (*binding_out).len() != (*binding_in).len() {
+        return Err(Error::malformed_control(
+            "unmatching binding sizes".to_string(),
+        ));
+    }
+
+    binding_out.iter().zip(binding_in.iter()).try_for_each(
+        |((k_out, v_out), (k_in, v_in))| {
+            if k_out.eq(k_in) && (v_out == v_in) {
+                Ok(())
+            } else {
+                Err(Error::malformed_control(
+                    "unmatching binding values for fed_in external cell"
+                        .to_string(),
```

Comment:
Better error message needed explaining what is wrong here. It might be worth using more descriptive names for variables than `k_out, v_out` etc. Then, the error message should say exactly what is wrong using a `format!` string.